testutil/daemon/daemon.go

```diff
@@ -519,12 +524,7 @@ out2:
 		return err
 	}
 
-	d.cmd.Wait()
-
-	if d.pidFile != "" {
-		return os.Remove(d.pidFile)
-	}
-	return nil
+	return d.cmd.Wait()
```

Comment:
Not quite related to this patch (except see below), but I see that `StartWithLogFile()` already creates a goroutine that calls `d.cmd.Waits()` and sends its return to the channel, `d.Wait`.

Now, this code calls `d.cmd.Wait()` one more time, I wonder why, one wait should be just enough. Looking into git logs, it comes all the way from ancient history. Maybe it's time to fix it. I would remove this second wait.

Now that you're returning the second `d.cmd.Wait()` exit code, it might actually return an error (I haven't checked that, just a hunch) as it was already been waited for.

Reply:
done